### DISCUSSION TITLE
CAPE variables in RAP grib2 files

### DISCUSSION THREAD

**Discussion by tlg7** [NOAA-GSL/GSL-Regional-Model-Forum#33] [Q&A] [answered]
Opening up RAP files with cfgrib, I have 5 variables of CAPE listed:
    CAPE at the surface, I assume is surface based CAPE.
    CAPE with a level type of heightAboveGroundLayer, which should be 0-3 km CAPE (event though the height level in the file is listed at 0).
    And finally CAPE variables with a level type of pressureFromGroundLayer. These have pressure levels of 9,000, 18,000, and 25,500 (PA). 

I can find a record of these five variables here:  https://www.nco.ncep.noaa.gov/pmb/products/rap/rap.t00z.awp130pgrbf00.grib2.shtml

Within the RAP table I have found, only two values of CAPE are found (SBCAPE and MUCAPE). https://rapidrefresh.noaa.gov/GRIB2Table_NCEPrap_prs.txt
Within a HRRR table, 4 values of CAPE were found. https://rapidrefresh.noaa.gov/GRIB2Table_NCEPrap_prs.txt.

Since MUCAPE looks at the lowest 300-mb, I assume this is equivalent to the pressureFromGroundLayer CAPE at 25,500 pa in the grib2 file. 

My confusion lies with the pressureFromGroundLayer CAPE variables at 9,000 and 18,000 pa. 

I have found that the former variable's (at 9,000 pa) values are equivalent to the NCL label CAPE_P0_2L108_GLC0(0), which is the CAPE using highest average theta-e in six lowest 30 mb layers, according to the HRRR table. Likewise, I have found that the later variable's (at 18,000 pa) is equivalent to CAPE_P0_2L108_GLC0(1), which is listed as Mixed-layer CAPE (MLCAPE) using average theta-e of the three lowest 30 mb layers.

So i suppose my questions are thus:
Am I correct that pressureFromGroundLayer CAPE at 25,000 pa is MUCAPE? 
Why would the CAPE which looks at the 3 lowest 30 mb layers be labeled 18,000 pa and not 9,000 pa? Or is there a typo? Which variable looks at what?

**Comment by EricJames-NOAA** [comment, marked as answer]
Hi @tlg7, thanks for your question.  The labeling of the GRIB2 CAPE fields is confusing.  I have looked into the UPP code in some detail, and I can confirm that:

(1) Yes, the field labeled pressureFromGroundLayer CAPE at 25500 Pa is most unstable CAPE.  This is found by looking for the max theta-e in the column.
(2) I think there is an error in GRIB2Table_NCEPrap_prs.txt in that the CAPE labeled as 9000 Pa uses the mean parcel characteristics over the lowest 3 30 hPa layers, but the one labeled as 18000 Pa is a most unstable CAPE over the lowest 6 30 hPa layers.  

Hopefully this helps to answer your question...I can continue to dig into the UPP code if there are further questions.

Thanks
Eric

**Reply by tlg7** [reply]
Thank you for the response, @EricJames-NOAA. You've cleared this up for me. Is there a common name for the most unstable cape over the six lowest 30 hpa layers? If I have further questions, I will ask here. Thank you again for your help.

Best,
Thomas

**Comment by GeoffManikin-NOAA** [comment]
Just a couple of minor clarifications:
 - the CAPE at 25500 Pa is computed by lifting the model level in the lowest 300 hPa with the highest theta-e  (there is an encoding issue in which the maximum allowed value for that octet of the product definition section is 255)
 - the CAPE at 9000 Pa is computed, as Eric noted, by averaging the characteristics of the lowest three 30 hPa layers to create a "mixed layer"
 - the CAPE at 18000 Pa is computed by lifting the most unstable of those six 30 hPa layers near the ground

**Reply by tlg7** [reply]
Hi @GeoffManikin-NOAA, 

Thank you very much for the clarifications. I see that the terms "mixed layer" and "mean layer" sometimes get mixed around. Is there a common term used for the most unstable cape over the lowest six 30 hpa layers?

Best,

Thomas

**Comment by GeoffManikin-NOAA** [comment]
Hi Thomas-

       I can't say if there is a common term across the user community, but
we at NCEP often refer to it as the 0-180 mb "best" or "most unstable" CAPE.

                          -Geoff
